### DISCUSSION TITLE
ScadaBR 1.2 - New Release

### DISCUSSION THREAD

**Discussion by fabiodurao** [SCADA-LTS/Scada-LTS#1943] [Show and tell]
Hello Scada-LTS team,

We have been following this project closely and are happy with the status achieved so far and with the future prospects. At the same time, in our local community a user appeared who felt the need for some updates and decided to develop, what we call ScadaBR 1.2 . We make the code available on github as well, being a new repository, without much to compare with previous code. To access, go to [https://github.com/ScadaBR/ScadaBR/releases/tag/v1.2](https://github.com/ScadaBR/ScadaBR/releases/tag/v1.2)

Below is the list of improvements for this version, which we can merge with Scada-LTS in the future:

- New installers for Windows and Linux (including ARM architecture, used on the Raspberry Pi, for example);
- ScadaBR 1.2 is now compatible with Java/OpenJDK 8 and Tomcat 9;
- Redesigned Login and Help pages, with a more modern style;
- Added CSS themes in different colors: green, black, brown, purple, pink, orange and red;
- A new font is used in the ScadaBR interface: the DejaVu Sans font;
- The “Dynamic GIF” components in the Graphical Representation had their design updated;
- Multi Tabs - now it is possible to open more than one page in the same browser;
- A new “Static Image” component has been added to Graphical Representations;
- Inclusion of the Flatpickr library (calendars on the alarms page);
- Inclusion of the FUScaBR 2.0 library;
- The list of components to be added in Graphical Representations is in alphabetical order;
- The popup windows for editing components of the Graphical Representation also started to be opened closer to the position of the mouse cursor;
- "Link" component of the Graphical Representation now suggests relative URLs;
- The components “Flex Builder” and “Humidity and Temperature Sensor” have been removed;
- Some “documentation popups” were updated or created from scratch (in the case of alarm pages and user profiles);
- Improved Portuguese translation. Untranslated expressions were filled with their English equivalents, avoiding interface errors due to lack of translation strings;
- ScadaBR gained an initial translation into Italian;
- Event search now includes date filters and allows multiple selection of a greater number of criteria;
- The popup window for adding user comments to an event is now displayed in the center of the screen;
- The list of data sources is now presented in alphabetical order;
- When editing a created data point, the popup window will open closer to the mouse cursor position;
- Modbus Serial protocol has been corrected and improved;
- The standard serial communication library has been replaced. Now ScadaBR uses the NRJavaSerial library (RXTX fork);
- Now the Modbus Serial data source will try to reconnect to the Modbus device if communication is lost;
- Update soundmanager2 library (for HTML5 audio);
- Rhino Library updated to version 1.7.13, correcting problems in ScadaBR functions that involve “scripting”;
- Removal of the JQuery library;
- Removal of wz_jsgraphics library;
- Fixes related to sending emails, such as the absence of the mangoLogoMed.jpg file;
- Pay attention to Java.security for sending emails - it is necessary to remove the restriction to TLS 1.0 and 1.1;
- Fixed 5 security vulnerabilities such as sending malicious JSP code cloaked in image;
- Bug fixes present in the Virtual data source;
- Bug fix related to Derby database creation;
- End of support for older browsers (such as Internet Explorer 9 or earlier);

We provide a video with subtitles where we briefly show the news [https://youtu.be/iS0v5qs11mc](https://youtu.be/iS0v5qs11mc) and we also recorded a larger video showing all the news in detail [https://youtu.be/Nvw6C3SIaYs](https://youtu.be/Nvw6C3SIaYs) .
The videos are in Portuguese but I believe the technical part is understandable.

**Comment by SoftQ** [comment]
Hello Fabio!
It is really great to hear from You!  I hope we may join our efforts very soon as it really does not make sense to develop multiple forks.
We will try to adapt these changes that are visible in separate commits.

Thank You
Michał

**Reply by fabiodurao** [reply]
Hi Michał,

Good to see you too! We are not interested in opening forks in the project, the initiative for this version was spontaneous in the community by the user Celso, who was improving several features in the frontend, so much so that the initial name was ScadaBR 1.1 EF (enhanced frontend), in parallel there was another user (Wagner) developing an installer for linux, they got together and started doing something unified, when we knew the project, it was almost ready, they were just fixing the serial modbus part, which for us was very useful, so we decided to release this version as 1.2.

We have in common agreement that for the next improvements, we want to collaborate with the Scada-LTS project, however, there were demands that were screaming among our community as:

- End of flash support by browsers -> this mainly impacted on not having more alarm sounds;
- End of support for java 7 -> it was already difficult to work with linux as I can't find this version in repositories with a simple "sudo apt install ....";
- Lack of serial modbus -> this bug came with ScadaBR 1.1, which was a problem for beginning users who are mostly college students and are dying to integrate ScadaBR with a simple uno arduino (without using ethernet shield, because it is expensive around here) by serial modbus. The worst thing is that these users resorted to ScadaBR 1.0 that uses java 6 and tomcat 6, totally outdated and with several known security vulnerabilities;

These demands are the main ones, but on the other hand, the community is excited to use Scada-LTS, however, it runs into some problems, the first one is that the installation is not an easy task. As I have already given training on the use of ScadaBR to more than 400 system integrators, I know the user profile well: 90% of users are from the electrical and automation area, technicians and engineers and the other 10% are divided into mechanics. , process engineer and IT professionals. The availability of the project by docker, seems to me to be a robust and elegant solution, but for 95% of users in the community, this installation is extremely complex, not to mention that the latest releases came with breaks in docker, docker compose among some other issues, that we, and myself included in this medium, do not understand what is happening and why the installation does not work. Manual installation for linux users is a little smoother, but less than 30% of ScadaBR users are familiar with linux, yet some beginning linux users still have difficulty in this installation format where each package is installed separately.

In order for the project to reach a larger community of users and developers at the same time, it cannot be faced with the first step, which is the installation. I believe that our community will be able to help in the construction of .exe installation for windows like next, next and finish and one in .sh for linux.

As with SBR 1.2, if you manage to pay some attention to this serial modbus feature [https://github.com/SCADA-LTS/Scada-LTS/issues/1699](https://github.com/SCADA-LTS/Scada-LTS/issues/1699), it would be extremely important to leverage the adoption of Scada-LTS in universities and then expand the number of users, so we can conquer a community that can support the project
